sdk/monitor/azure-monitor-query/src/test/java/com/azure/monitor/query/LogsQueryAsyncClientTest.java

```diff
@@ -0,0 +1,338 @@
+        logsBatchQuery.addWorkspaceQuery(WORKSPACE_ID, QUERY_STRING + " | take", null);
+
+        StepVerifier.create(client
+                        .queryBatchWithResponse(logsBatchQuery, Context.NONE))
+                .assertNext(response -> {
+                    LogsBatchQueryResultCollection batchResultCollection = response.getValue();
+                    List<LogsBatchQueryResult> responses = batchResultCollection.getBatchResults();
+
+                    assertEquals(2, responses.size());
+                    assertEquals(LogsQueryResultStatus.SUCCESS, responses.get(0).getQueryResultStatus());
+                    assertNull(responses.get(0).getError());
+                    assertEquals(LogsQueryResultStatus.FAILURE, responses.get(1).getQueryResultStatus());
+                    assertNotNull(responses.get(1).getError());
+                    assertEquals("BadArgumentError", responses.get(1).getError().getCode());
+                })
+                .verifyComplete();
+    }
+
+    @Test
+    public void testStatistics() {
+        StepVerifier.create(client.queryWorkspaceWithResponse(WORKSPACE_ID,
+                        QUERY_STRING, null, new LogsQueryOptions().setIncludeStatistics(true), Context.NONE))
+                .assertNext(response -> {
+                    LogsQueryResult queryResults = response.getValue();
+                    assertEquals(1, queryResults.getAllTables().size());
+                    assertNotNull(queryResults.getStatistics());
+                })
+                .verifyComplete();
+    }
+
+    @Test
+    public void testBatchStatistics() {
+        LogsBatchQuery logsBatchQuery = new LogsBatchQuery();
+        logsBatchQuery.addWorkspaceQuery(WORKSPACE_ID, QUERY_STRING, null);
+        logsBatchQuery.addWorkspaceQuery(WORKSPACE_ID, QUERY_STRING, null,
+                new LogsQueryOptions().setIncludeStatistics(true));
+
+        StepVerifier.create(client
+                        .queryBatchWithResponse(logsBatchQuery, Context.NONE))
+                .assertNext(response -> {
+                    LogsBatchQueryResultCollection batchResultCollection = response.getValue();
+                    List<LogsBatchQueryResult> responses = batchResultCollection.getBatchResults();
+
+                    assertEquals(2, responses.size());
+                    assertEquals(LogsQueryResultStatus.SUCCESS, responses.get(0).getQueryResultStatus());
+                    assertNull(responses.get(0).getError());
+                    assertNull(responses.get(0).getStatistics());
+                    assertEquals(LogsQueryResultStatus.SUCCESS, responses.get(1).getQueryResultStatus());
+                    assertNull(responses.get(1).getError());
+                    assertNotNull(responses.get(1).getStatistics());
+                })
+                .verifyComplete();
+    }
+
+    @Test
+    @DisabledIfEnvironmentVariable(named = "AZURE_TEST_MODE", matches = "LIVE", disabledReason = "server timeout is "
```

Comment:
Should this also be disabled for RECORD? Would it be better for this to use `DisableIf` and use a shareable function between sync and async with more control: https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/condition/DisabledIf.html

Reply:
No, for record, we can still run it with some amount of retries locally to eventually succeed and get the recordings.